compiler/semtempl.nim

```diff
@@ -372,7 +372,11 @@ proc semTemplBody(c: var TemplCtx, n: PNode): PNode =
     openScope(c)
     n.sons[L-2] = semTemplBody(c, n.sons[L-2])
     for i in countup(0, L - 3):
-      addLocalDecl(c, n.sons[i], skForVar)
+      if n[i].kind == nkVarTuple:
+        for j in 0 ..< sonsLen(n[i])-1:
```

Comment:
``..<`` AND -1?  shouldn't this be ``n[i].len - 3``? (Don't process the type decl nor the value)

Reply:
> shouldn't this be n[i].len - 3
  
If i understood the code correctly, then it is trying to emit a local declaration for each loop variable (skForVar). In that case, I think my code is correct. Also the outer for loop runs from 0 to L-3, so it does not process what it should not.
  
> `..<` AND -1
  
`nkVarTuple` for for loops has an empty node at the end (because we reuse the code for var/const tuple parsing). And all previous nodes contains the variables for the unpacking.